python/cudf/cudf/pylibcudf_tests/test_hashing.py

```diff
@@ -0,0 +1,186 @@
+
+
+@pytest.fixture
+def pa_input_column(pa_type):
+    if pa.types.is_integer(pa_type) or pa.types.is_floating(pa_type):
+        return pa.array([1, 2, 3], type=pa_type)
+    elif pa.types.is_string(pa_type):
+        return pa.array(["a", "b", "c"], type=pa_type)
+    elif pa.types.is_boolean(pa_type):
+        return pa.array([True, True, False], type=pa_type)
+    elif pa.types.is_list(pa_type):
+        # TODO: Add heterogenous sizes
+        return pa.array([[1], [2], [3]], type=pa_type)
+    elif pa.types.is_struct(pa_type):
+        return pa.array([{"v": 1}, {"v": 2}, {"v": 3}], type=pa_type)
+    raise ValueError("Unsupported type")
+
+
+@pytest.fixture()
+def input_column(pa_input_column):
+    return plc.interop.from_arrow(pa_input_column)
+
+
+@pytest.fixture(scope="module")
+def list_struct_table():
+    data = pa.Table.from_pydict(
+        {
+            "list": [[1, 2, 3], [4, 5, 6], [7, 8, 9]],
+            "struct": [{"a": 1, "b": 2}, {"a": 3, "b": 4}, {"a": 5, "b": 6}],
+        }
+    )
+    return data
+
+
+def python_hash_value(x, method):
+    if isinstance(x, str):
+        binary = str(x).encode()
+    elif isinstance(x, float):
+        binary = struct.pack("<d", x)
+    elif isinstance(x, bool):
+        binary = x.to_bytes(1, byteorder="little", signed=True)
+    elif isinstance(x, int):
+        binary = x.to_bytes(8, byteorder="little", signed=True)
+    elif isinstance(x, list):
+        binary = np.array(x).tobytes()
+    else:
+        raise NotImplementedError
+    if method == "murmurhash3_x86_32":
+        # reimplement libcudf hash combine for a single colum
+        seed = plc.hashing.LIBCUDF_DEFAULT_HASH_SEED
+        hashval = mmh3.hash(binary, seed)
+        return seed ^ (hashval + 0x9E3779B9 + (seed << 6) + (seed >> 2))
+    elif method == "murmurhash3_x64_128":
+        hasher = mmh3.mmh3_x64_128(seed=plc.hashing.LIBCUDF_DEFAULT_HASH_SEED)
+        hasher.update(binary)
+        # libcudf returns a tuple of two 64-bit integers
+        return hasher.utupledigest()
+    elif method == "xxhash_64":
+        return xxhash.xxh64(
+            binary, seed=plc.hashing.LIBCUDF_DEFAULT_HASH_SEED
+        ).intdigest()
+    else:
+        return getattr(hashlib, method)(binary).hexdigest()
+
+
+@pytest.mark.parametrize(
+    "method", ["sha1", "sha224", "sha256", "sha384", "sha512"]
+)
+def test_hash_column_sha(pa_input_column, method):
+    plc_tbl = plc.interop.from_arrow(
+        pa.Table.from_arrays([pa_input_column], names=["data"])
+    )
+    plc_hasher = getattr(plc.hashing, method)
+
+    if isinstance(pa_input_column.type, (pa.ListType, pa.StructType)):
+        with pytest.raises(TypeError):
+            plc_hasher(plc_tbl)
+        return
+
+    expect = pa.array(
+        [
+            python_hash_value(val, method)
+            for val in pa_input_column.to_pylist()
+        ],
+        type=pa.string(),
+    )
+    got = plc_hasher(plc_tbl)
+    assert_column_eq(got, expect)
```

Comment:
It seems like we're using arguments in `assert_column_eq` in a swapped order vs. how we use `assert_eq` in cudf tests. There _is_ a correct direction for this, because it affects the way that errors are printed. I believe we need to swap this and do a follow-up PR to fix [the other instances](https://github.com/search?q=repo%3Arapidsai%2Fcudf+assert_column_eq%28&type=code) of `assert_column_eq(got, expect)`.
```suggestion
    assert_column_eq(expect, got)
```

Reply:
Right now `assert_column_eq` isn't symmetric and we're required to pass the pylibcudf array as the first argument. It probably could be symmetrized though. I think it's a separate change, but one I'm happy to make before this PR goes in.

Reply:
Thanks! To clarify: it’s okay that it’s asymmetric. We just want to be consistent about (expected, actual) vs. (actual, expected) across all our equality testing utilities.